src/i2c.rs

```diff
@@ -108,6 +210,48 @@ pub enum Error {
     Arbitration,
     /// NACK received
     NotAcknowledge,
+    /// Target operation only:
+    /// Indicates that a stop or repeat start was received while reading, or
+    /// while explicitly waiting for a controller read or write event.
+    TransferStopped,
+    /// Target operation only:
+    /// While waiting for a controller read event, a write event was received.
+    ControllerExpectedWrite,
+    /// Target operation only:
+    /// While waiting for a controller write event, a read event was received.
+    ControllerExpectedRead,
+}
+
+/// Target Event.
+///
+/// This encapsulates a transaction event that occurs when listening in Target
+/// operation. A TargetWrite event indicates that the Controller wants to read
+/// data from the Target (I2C read operation). A TargetRead event indicates
+/// that the Controller wants to write data to the Target (I2C write
+/// operation). A Stop event indicates that a Stop condition was received and
+/// the transaction has been completed.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum TargetEvent {
+    /// The controller initiated an I2C Read operation, requiring that the Target must write to the
+    /// bus.
+    TargetWrite { address: u16 },
+    /// The controller initiated an I2C Write operation, requiring that the Target must read from
+    /// the bus.
+    TargetRead { address: u16 },
+    /// A Stop condition was received, ending the transaction.
+    Stop,
+}
+
+/// Target Listen Event
+///
+/// Indicates what listen events are responded to. A target can respond to one
+/// of or all of a general call address, the primary address configured or a
+/// secondary address.
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum TargetListenEvent {
+    GeneralCall,
```

Comment:
I'd document what these are supposed to be. I get primary/secondary address, but GeneralCall is pretty opaque.